src/components/Search/SearchPageHeaderInput.tsx

```diff
@@ -0,0 +1,273 @@
+
+            setAutocompleteSubstitutions(substitutions);
+        }
+    };
+
+    const hideAutocompleteList = () => setIsAutocompleteListVisible(false);
+    const showAutocompleteList = () => setIsAutocompleteListVisible(true);
+
+    if (isCannedQuery) {
+        const headerIcon = getHeaderContent(type).icon;
+
+        return (
+            <View
+                dataSet={{dragArea: false}}
+                style={[styles.headerBar, isCannedQuery && styles.headerBarDesktopHeight]}
+            >
+                <View style={[styles.dFlex, styles.flexRow, styles.alignItemsCenter, styles.flexGrow1, styles.justifyContentBetween, styles.overflowHidden]}>
+                    <Icon
+                        src={headerIcon}
+                        width={variables.iconHeader}
+                        height={variables.iconHeader}
+                        additionalStyles={[styles.mr2]}
+                    />
+                    <Header subtitle={<Text style={[styles.textLarge, styles.textHeadlineH2]}>{headerText}</Text>} />
+                    <View style={[styles.reportOptions, styles.flexRow, styles.pr5, styles.alignItemsCenter, styles.gap2]}>
+                        {children}
+                        <SearchButton />
+                    </View>
+                </View>
+            </View>
+        );
+    }
+
+    const searchQueryItem = textInputValue
+        ? {
+              text: textInputValue,
+              singleIcon: Expensicons.MagnifyingGlass,
+              searchQuery: textInputValue,
+              itemStyle: styles.activeComponentBG,
+              keyForList: 'findItem',
+              searchItemType: CONST.SEARCH.SEARCH_ROUTER_ITEM_TYPE.SEARCH,
+          }
+        : undefined;
+
+    const isHeaderInputActive = isAutocompleteListVisible;
+
+    // we need `- BORDER_WIDTH` to achieve the effect that the input will not "jump"
+    const popoverHorizontalPosition = 12 - BORDER_WIDTH;
+    const autocompleteInputStyle = isHeaderInputActive
+        ? [
+              styles.border,
+              styles.borderRadiusComponentLarge,
+              styles.pAbsolute,
+              styles.pt2,
+              {top: 8 - BORDER_WIDTH, left: popoverHorizontalPosition, right: popoverHorizontalPosition},
+              {boxShadow: variables.popoverMenuShadow},
+          ]
+        : [styles.pt4];
+    const inputWrapperStyle = isHeaderInputActive ? styles.ph2 : null;
+
+    return (
+        <View
+            dataSet={{dragArea: false}}
+            style={[styles.searchResultsHeaderBar, styles.mh85vh, isHeaderInputActive && styles.ph3]}
+        >
+            <View style={[styles.appBG, ...autocompleteInputStyle]}>
+                <SearchRouterInput
+                    value={textInputValue}
+                    onSearchQueryChange={onSearchQueryChange}
+                    isFullWidth
+                    onSubmit={() => {
+                        submitSearch(textInputValue);
+                    }}
```

Comment:
This callback should be called only if no item is focused. Otherwise on pressing Enter two handlers are executed: `onSubmit` (this one)  and the `onListItemPress` which conflicts